awx/main/apps.py

```diff
@@ -34,7 +36,30 @@ def load_named_url_feature(self):
             category_slug='named-url',
         )
 
+    def _load_credential_types_feature(self):
+        """
+        Create CredentialType records for any discovered credentials.
+
+        Note that Django docs advise _against_ interacting with the database using
+        the ORM models in the ready() path. Specifically, during testing.
+        However, we explicitly use the @bypass_in_test decorator to avoid calling this
+        method during testing.
+
+        Django also advises against running pattern because it runs everywhere i.e.
+        every management command. We use an advisory lock to ensure correctness and
+        we will deal performance if it becomes an issue.
+        """
+        from awx.main.models.credential import CredentialType
+
+        if is_database_synchronized():
+            CredentialType.setup_tower_managed_defaults(app_config=self)
+
+    @bypass_in_test
+    def load_credential_types_feature(self):
+        return self._load_credential_types_feature()
+
     def ready(self):
         super().ready()
 
+        self.load_credential_types_feature()
```

Comment:
would it be possible to condition this to specific awx-manage commands instead all? this way it runs less often

Reply:
Nope. If we want it to run as part of another management command then we should put it in that management command. Adding something like the below would be much more confusing.

```
if 'run_dispatcher' in ' '.join(sys.argv):
  self.load_credential_types_feature()
```